### PR TITLE
Try to remove a few LogicErrors and DeadStores in DT/CSC LocalMuon reco

### PR DESCRIPTION
**Pull request by perrotta**
#### PR description:

As the PR title says
Issues were pointed out by the static analyzer during standard tests

#### PR validation:

It builds and successfully runs the short matrix
No differences expected in output (excepted perhaps tiny effects due to the numerical precision of the replacement of the constant sqrt(12) with 1/sqrt(12) in CSCMake2DRecHit.cc, which can be reverted anyhow, @ptcox)

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@namapane @fcavallo @battibass FYI

**Comment by perrotta** [comment]
@ptcox (for CSC) and @namapane @fcavallo @battibass (for DT): could you please give your green light, or let me know when you have time to do so?

**Comment by namapane** [comment]
@perrotta, I had a look to DT-related changes, and could not spot anything wrong.

**Comment by ptcox** [comment]
OK by me, as long as the comment and the 't_zero' in the output text, 
remain.

Tim

Slava Krutelyov wrote on 8/29/19 16:12:

**Comment by perrotta** [comment]
+1
- A few fixes and cleanings in CSC and DT local reco packages started with the intent of getting rid of a few warnings raised by the static analyzer
- jenkins tests pass and show no differences, as they should

**Comment by fabiocos** [comment]
please test

**Comment by battibass** [comment]
@perrotta : apologies for the delay, I had a look and it seems OK to me as well.

File: RecoLocalMuon/CSCRecHitD/src/CSCMake2DRecHit.cc [hunk 1]
```diff
@@ -133,8 +133,7 @@ CSCRecHit2D CSCMake2DRecHit::hitFromStripAndWire(const CSCDetId& id,
   }
   tpeak = peakTimeFinder_->peakTime(tmax, adcArray, tpeak);
   // Just for completeness, the start time of the pulse is 133 ns earlier, according to Stan :)
-  float t_zero = tpeak - 133.f;
-  LogTrace("CSCRecHit") << "[CSCMake2DRecHit] " << id << " strip=" << centerStrip << ", t_zero=" << t_zero
+  LogTrace("CSCRecHit") << "[CSCMake2DRecHit] " << id << " strip=" << centerStrip << ", t_zero=" << tpeak - 133.f
```
[the remark this answers is not in the document]
Comment (fabiocos) [reply RecoLocalMuon/CSCRecHitD/src/CSCMake2DRecHit.cc]: @slava77 after the integration of https://github.com/cms-sw/cms-bot/pull/1197 this flag is enabled, as far as I can see in the latest test I've run https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ecd997/2318/llvm-analysis/ there is no new claim of unused variable. So are you ok with this update, or would you suggest to revert it?

[the remark this answers is not in the document]
Comment (slava77) [reply RecoLocalMuon/CSCRecHitD/src/CSCMake2DRecHit.cc]: I'm fine for this simple case.

[the remark this answers is not in the document]
Comment (fabiocos) [reply RecoLocalMuon/CSCRecHitD/src/CSCMake2DRecHit.cc]: thanks, this PR will be integrated in next IB

**Comment by fabiocos** [comment]
+1

**Pull request merged**